x-pack/plugins/data_enhanced/public/search/async_search_strategy.ts

```diff
@@ -53,8 +53,13 @@ export const asyncSearchStrategyProvider: TSearchStrategyProvider<typeof ASYNC_S
 
       return search(request, options).pipe(
         expand(response => {
+          // If the response indicates of an error, stop polling and complete the observable
+          if (!response || (response.is_partial && !response.is_running)) {
+            return throwError(new AbortError());
+          }
+
           // If the response indicates it is complete, stop polling and complete the observable
-          if ((response.loaded ?? 0) >= (response.total ?? 0)) return EMPTY;
+          if (!response.is_running) return EMPTY;
```

Comment:
Would be nice to add a unit test case that captures this bug.

Reply:
Done!